src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/clang.stack_protector.a.sarif

```diff
@@ -42,12 +42,13 @@
       "artifacts": [
         {
           "location": {
-            "uri": "file:///Z:/src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/clang.stack_protector.a"
+            "uri": "file:///Z:/src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/clang.stack_protector.a",
+            "index": 0
           },
           "hashes": {
-            "md5": "2400EE99A6BB7B792A67A03A59385018",
-            "sha-1": "525CB4CBA8B8F80539BBC86C2D78A4275279D8F6",
-            "sha-256": "41BF336F8AB7D9A6246D37A6EC7210D4BE07893836890BAB0BFEDA98ACEDE87D"
+            "md5": "C60E01E6A2660A3359EACF5D00799FA0",

```

Comment:
>C60E01E6A2660A3359EACF5D00799FA0 [](http://example.com/codeflow?start=20&length=32)

@shaopeng, can you explain why these file hashes changed when I rebaselined?

Reply:
a first glance these were created by Everett Maus not me,
looking...

Reply:
I checked both files in your branch. The correct hashes are the original ones...

Reply:
I suspect this is wrong
to get hash we now use:
`fileText = _fileSystem.FileReadAllText(path);`
and it was:
`Stream stream = fileSystem.FileOpenRead(fileName)`
a binary should not read by text?

Reply:
@michaelcfanning @suvamM @schlaman-ms 
found the changeset that I mentioned at the end of meeting, this changeset have a few open comments from Michael, this particular one I looked and replied.
Do we think it is a issue,
does it need to be addressed within 4.1.0?

Reply:
Thanks for recalling this issue!! Yes, we need to address this. We shouldn't be changing the hashing mechanism for binskim. Isn't this what we need?

  using (Stream stream = fileSystem.FileOpenRead(fileName))

Reply:
my understanding of the problem is, we have added a new cache `FileRegionsCache`, and that cache is used for something else, but since we have it, we also re-use it for hashing purpose, but because it is string, we can not it will be different value after hash.
So, should be fix be, change the cache to store binary, then the other place that we need string if we can read string from the binary then the cache is re-usable.

or, another way is just not reuse the string cache, when hashing read from the disk again.